core/src/memory.rs

```diff
@@ -158,14 +158,14 @@ impl PageNumber {
 impl core::ops::Add<PageNumber> for PageNumber {
     type Output = Self;
     fn add(self, other: Self) -> Self {
-        Self(self.0 + other.0)
+        Self(self.0.saturating_add(other.0))
```

Comment:
`impl Add` which does `saturating_add` is really counter-intuitive and should be avoided